abb_driver/rapid/ROS_motion.mod

```diff
@@ -47,25 +49,33 @@ PROC main()
 
     WHILE true DO
         ! Check for new Trajectory
-        IF (ROS_new_trajectory)
+        IF (ROS_new_trajectory) THEN
+            TPWrite "new trajectory!";
             init_trajectory;
+        ENDIF
 
         ! execute all points in this trajectory
         IF (trajectory_size > 0) THEN
+            TPWrite "executing trajectory";
             FOR current_index FROM 1 TO trajectory_size DO
                 target.robax := trajectory{current_index}.joint_pos;
-                target.extax := trajectory{current_index}.extax_pos;
 
-                skip_move := (current_index = 1) AND is_near(target, 0.1, 0.1);
+                ! TODO(bhomberg): include rail position in this calculation
+                skip_move := (current_index = 1) AND is_near(target.robax, 0.1);
 
                 stop_mode := DEFAULT_CORNER_DIST;  ! assume we're smoothing between points
-                IF (current_index = trajectory_size) stop_mode := fine;  ! stop at path end
-
+                IF (current_index = trajectory_size) THEN
+                    stop_mode := fine;  ! stop at path end
+                ENDIF
                 ! Execute move command
-                IF (NOT skip_move)
+                IF (NOT skip_move) THEN
+		            FESTO_rail_position := trajectory{current_index}.rail_position;
+                    ! TODO(bhomberg): set velocity to max?
+                    FESTO_rail_velocity := 100;

```

Comment:
Can you read that somehow from the trajectory (distance / expected duration)?

Reply:
That's my eventual plan, but it's not particularly straightforward, unfortunately -- we set the velocity between 0% and 100% of the configured max velocity number buried deep in config files.  Ideally we'd access what that value is somehow (and at the very least I need to go look up what it's currently set to).  I can add a TODO, though.